Userland/Applications/IRCClient/IRCWindow.cpp

```diff
@@ -178,6 +178,7 @@ IRCWindow::IRCWindow(IRCClient& client, void* owner, Type type, const String& na
             m_client->handle_user_input_in_server(m_text_box->text());
         m_text_box->add_current_text_to_history();
         m_text_box->clear();
+        m_text_box->allow_undo(false);
```

Comment:
This feels very strange. So after one message, you disable the undo action, but all the undo state is still there. If the undo action becomes re-enabled in the future, it will again be possible to go backwards to the old state.

What we really want is to wipe the entire undo stack of the `GUI::TextDocument` after sending.

The code would look something like:
```cpp
m_text_box->document().undo_stack().clear();
```

Reply:
`If the undo action becomes re-enabled in the future, it will again be possible to go backwards to the old state.` Yes. The current change allows you to undo back to the previous message's text. But it never crashed in all the possible scenarios I tried. Though not the most correct way to do it, I agree.

`What we really want is to wipe the entire undo stack of the GUI::TextDocument after sending.` This is exactly what I think would be the best (and the most correct) way as well.

I have implemented the change you suggested. Please have a look at it. Thank you.